### PR TITLE
Refactor core-setup dotnethost packaging

### PR DESCRIPTION
**Pull request by chcosta**
Refactors packaging, aligning it with how we package in corefx, coreclr, and even in core-setup previously for the Microsoft.NETCore.App package.  

@ellismg @ericstj

### CODE REVIEW

**Comment by ericstj** [comment]
I think @ramarag was considering deleting all the host packages.

**Comment by chcosta** [comment]
Looking into the failures.

**Comment by gkhanna79** [comment]
CC @eerhardt

**Comment by ellismg** [comment]
I need this in order to be able to build Core-Setup for a RID we've never seen before (since the packaging stuff ensures we always generate a run-time package for the current platform and add it to the lineup).

Even If these packages are going away eventually, I'd still prefer us to merge this in the interm as it unblocks build from source work.

File: pkg/pack.cmd [hunk 1]
```diff
@@ -20,7 +20,7 @@ if exist "%__ProjectDir%\bin" (rmdir /s /q "%__ProjectDir%\bin")
 if not ERRORLEVEL 0 goto :Error
 
 :: Package the assets using Tools
-"%__DotNet%" "%__MSBuild%" "%__ProjectDir%\projects\packages.builds" /p:TargetsWindows=true /verbosity:minimal
+"%__DotNet%" "%__MSBuild%" "%__ProjectDir%\projects\packages.builds" /p:OSGroup=Windows_NT /verbosity:minimal /flp:v=diag
```
Comment (ellismg) [inline pkg/pack.cmd]: Can you either remove /flp:v=diag or also add it to the list of arguments in pack.sh?

File: pkg/projects/Microsoft.NETCore.DotNetHost/Microsoft.NETCore.DotNetHost.pkgproj [hunk 2]
```diff
@@ -4,70 +4,9 @@
 
   <PropertyGroup>
     <Version>$(HostVersion)</Version>
-    <SkipPackageFileCheck>true</SkipPackageFileCheck>
-    <SkipValidatePackage>true</SkipValidatePackage>
-    <PackagePlatforms>x64;x86;arm;arm64;armel;</PackagePlatforms>
-    <OutputPath>$(PackagesOutputPath)</OutputPath>
-    <IncludeRuntimeJson>true</IncludeRuntimeJson>
-    <Serviceable>true</Serviceable>
```
Comment (gkhanna79) [inline pkg/projects/Microsoft.NETCore.DotNetHost/Microsoft.NETCore.DotNetHost.pkgproj]: Is the serviceable attribute getting propagated somehow in the new model?

Comment (ellismg) [inline pkg/projects/Microsoft.NETCore.DotNetHost/Microsoft.NETCore.DotNetHost.pkgproj]: It is set in `pkg/projects/dir.props` which all of these things import.

Comment (chcosta) [inline pkg/projects/Microsoft.NETCore.DotNetHost/Microsoft.NETCore.DotNetHost.pkgproj]: What Matt said was the intention, but you reminded me that Microsoft.NETCore and Microsoft.NETCore.UniversalWindowsPlatform  bypass importing that dir.props, so I moved it up a level into `pkg/dir.props`

File: pkg/projects/dir.props [hunk 3]
```diff
@@ -1,11 +1,123 @@
...
+    <OfficialBuildRID Include="fedora.24-x64" />
+    <OfficialBuildRID Include="linux-x64" />
+    <OfficialBuildRID Include="opensuse.42.1-x64" />
+    <OfficialBuildRID Include="osx.10.10-x64" />
+    <OfficialBuildRID Include="rhel.7-x64" />
+    <OfficialBuildRID Include="ubuntu.14.04-x64" />
+    <OfficialBuildRID Include="ubuntu.16.04-x64" />
+    <OfficialBuildRID Include="ubuntu.16.10-x64" />
+    <OfficialBuildRID Include="win7-x86">
+      <BuidOnRID>win10-x86</BuidOnRID>
```
Comment (gkhanna79) [inline pkg/projects/dir.props]: What is BuildOnRID used for?

File: pkg/projects/dir.props [hunk 4]
```diff
@@ -1,11 +1,123 @@
...
+      <Platform>x86</Platform>
+    </OfficialBuildRID>
+    <OfficialBuildRID Include="win7-x64">
+      <BuidOnRID>win10-x64</BuidOnRID>
+    </OfficialBuildRID>
+    <OfficialBuildRID Include="win8-arm">
+      <BuidOnRID>win10-arm</BuidOnRID>
```
Comment (gkhanna79) [inline pkg/projects/dir.props]: win10-arm does not sound right.

File: pkg/projects/dir.props [hunk 5]
```diff
@@ -1,11 +1,123 @@
...
+      <Platform>arm</Platform>
+    </OfficialBuildRID>
+    <OfficialBuildRID Include="win10-x86">
+      <Platform>x86</Platform>
+    </OfficialBuildRID>
+    <OfficialBuildRID Include="win10-x64" />
+    <OfficialBuildRID Include="win10-arm">
```
Comment (gkhanna79) [inline pkg/projects/dir.props]: Is win10-arm a new RID?

File: pkg/projects/dir.props (hunk 4 above)
Comment (chcosta) [inline pkg/projects/dir.props]: I didn't change that, it was present in `https://github.com/dotnet/core-setup/blob/master/pkg/projects/Microsoft.NETCore.App/Microsoft.NETCore.App.props#L29`.  I think it's a way of saying to build that `OfficialBuildRid` even if you're on a different rid.  @ericstj

File: pkg/projects/dir.props (hunk 5 above)
Comment (chcosta) [inline pkg/projects/dir.props]: Same comment as above.  I didn't want to change what was working.

File: pkg/projects/Microsoft.NETCore.UniversalWindowsPlatform/Microsoft.NETCore.UniversalWindowsPlatform.builds [hunk 6]
```diff
@@ -1,10 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <Import Project="..\..\dir.props" />
```
Comment (eerhardt) [inline pkg/projects/Microsoft.NETCore.UniversalWindowsPlatform/Microsoft.NETCore.UniversalWindowsPlatform.builds]: Why this change?  I think the directory structure is broken if we are "skipping" dir.props files in the hierarchy.  It makes it awfully confusing.

File: pkg/projects/Microsoft.NETCore.UniversalWindowsPlatform/Microsoft.NETCore.UniversalWindowsPlatform.builds [hunk 7]
```diff
@@ -1,10 +1,10 @@
...
 
   <ItemGroup>
     <Project Include="Microsoft.NETCore.UniversalWindowsPlatform.pkgproj" />
   </ItemGroup>
 
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.traversal.targets))\dir.traversal.targets" />
+  <Import Project="..\..\dir.traversal.targets" />
```
Comment (eerhardt) [inline pkg/projects/Microsoft.NETCore.UniversalWindowsPlatform/Microsoft.NETCore.UniversalWindowsPlatform.builds]: You didn't make this change in pkg/projects/Microsoft.NETCore.UniversalWindowsPlatform/Microsoft.NETCore.UniversalWindowsPlatform.pkgproj.  Was that intentional?

File: pkg/projects/packages.builds [hunk 8]
```diff
@@ -1,10 +1,10 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <Import Project="..\dir.props" />
```
Comment (eerhardt) [inline pkg/projects/packages.builds]: I don't understand this change.  Is it necessary?

File: pkg/projects/dir.props [hunk 9]
```diff
@@ -1,11 +1,121 @@
...
+
+    <IsLineupPackage Condition="'$(PackageTargetRuntime)' == ''">true</IsLineupPackage>
   </PropertyGroup>
+
+  <PropertyGroup Condition="'$(PackageTargetRuntime)' == ''">
+    <SkipValidatePackage>true</SkipValidatePackage>
+    <IncludeRuntimeJson>true</IncludeRuntimeJson>
+  </PropertyGroup>
+
+  <Choose>
```
Comment (eerhardt) [inline pkg/projects/dir.props]: This whole chunk is only used in a single project.  Does it need to be brought into the common dir.props?

File: pkg/projects/dir.props [hunk 10]
```diff
@@ -1,11 +1,121 @@
...
+        <LibraryFileExtension>.so</LibraryFileExtension>
+        <SymbolFileExtension>.dbg</SymbolFileExtension>
+      </PropertyGroup>
+    </Otherwise>
+  </Choose>
+
+  <PropertyGroup Condition="'$(PackageRID)' == ''">
+    <PackageRID>$(DistroRid)</PackageRID>
+    <PackageRID Condition="'$(OSEnvironment)' == 'OSX'">osx.10.10-$(Platform)</PackageRID>
+    <PackageRID Condition="'$(OSEnvironment)' == 'Windows_NT'">win10-$(Platform)</PackageRID>
```
Comment (eerhardt) [inline pkg/projects/dir.props]: These two lines seem odd to me.  For OSX we are defaulting to the lowest supported RID.  For Windows we default to the highest supported RID.

File: pkg/projects/Microsoft.NETCore.DotNetHostResolver/Microsoft.NETCore.DotNetHostResolver.props [hunk 11]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <Version>$(HostResolverVersion)</Version>
```
Comment (eerhardt) [inline pkg/projects/Microsoft.NETCore.DotNetHostResolver/Microsoft.NETCore.DotNetHostResolver.props]: I find it odd that the `DotNetHostResolver` package is the only one that has this file. It isn't in the HostPolicy, nor the two Host packages.  Why is this one special?

I looked really quick, but I don't see where this file gets imported.
f
f

File: pkg/projects/dir.props (hunk 9 above)
Comment (chcosta) [inline pkg/projects/dir.props]: There is other code in a separate PR that will make use of this code.  It's not really a project specific thing.

File: pkg/projects/dir.props (hunk 10 above)
Comment (chcosta) [inline pkg/projects/dir.props]: This was pre-existing.

File: pkg/projects/Microsoft.NETCore.UniversalWindowsPlatform/Microsoft.NETCore.UniversalWindowsPlatform.builds (hunk 6 above)
Comment (chcosta) [inline pkg/projects/Microsoft.NETCore.UniversalWindowsPlatform/Microsoft.NETCore.UniversalWindowsPlatform.builds]: It's confusing because we had a system where only one project used the new packaging model (Microsoft.NETCore.App), and it self-contained everything, then I moved that into a common location, but there are still two packages that don't produce runtime packages and are on the old model.  I think I can try to make them all play nicely though.

File: pkg/projects/packages.builds (hunk 8 above)
Comment (chcosta) [inline pkg/projects/packages.builds]: It was because the new model attempts to add a project reference to the pkgproj for each builds file.  We don't hit this in other repo's because it's not all crammed into the same area.  I can clean it up.

**Comment by gkhanna79** [comment]
CC @weshaggard

File: pkg/packages.builds [hunk 12]
```diff
@@ -3,7 +3,7 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
```
Comment (weshaggard) [inline pkg/packages.builds]: Any particular reason for the moving of this file?

Comment (chcosta) [inline pkg/packages.builds]: This was the simplest solution since `pkg\projects\dir.props` and `pkg\projects\dir.traversal.targets` aren't intended to apply to this file and @eerhardt wasn't happy with special casing the imports to `..\dir.blah`.  We don't have to deal with this in other repos where packages.builds is also not a part of the same folder layout which handles buildrids / filtering.

**Comment by chcosta** [comment]
@ramarag , do you have any concerns with this change?

**Comment by ramarag** [comment]
@chcosta i don't have any concerns from the cursory look
